pages/index.vue

```diff
@@ -64,19 +65,22 @@ export default Vue.extend({
         case 'latestmedia': {
           const latestMediaSections = [];
 
-          const userViewsRequest = await $api.userViews.getUserViews({
-            userId: $auth.user?.Id
-          });
+          let userViews: BaseItemDto[] = store.state.userViews.views;
+
+          if (!userViews.length) {
+            await store.dispatch('userViews/refreshUserViews');
+            userViews = await store.state.userViews.views;
+          }
```

Comment:
What about mapping the state of userViews to the computed properties, set an immediate and deep watcher that checks if the length is 0 and, if not, dispatches the action?

This would easily allows us in the future to listen to websocket messages globally and dispatch refresh actions for each of the stores as soon as updates are received. With this approach, nothing is reactive

Reply:
Unfortunately we can't because we are accessing it in a sync data.

Reply:
That's not what I meant. Let me explain better (as I also noticed I mentioned using a watcher when we don't really need it):

My proposed approach:

1. ``mapState`` views from store and replace ``userViews`` variable with it.
2. In store action, do the api request and check with lodash's ``isEqual`` if the response is equal to what we have in the store. If it's equal, do nothing, if it isn't, commit a mutation, so the mappedState in point 1 gets an update and user views are changed to the user.
3. In ``asyncData``, dispatch ``refreshUserViews`` action.

With this, we get the following:

1. Cleaner
2. Fully reactive if we commit the refresh action on a websocket message somewhere else after initialization
3. ¿Perhaps better load times as the component will have the store populated right before ``asyncData``?

With the proposed schema, I also wonder if we should get rid of asyncData here, as the page loading will be blocked until the store action is completed (which might be pointless since there can be the case where the response equals to what we already had in the store). Maybe remove ``await`` from the action so the hook doesn't wait for it's completion?

Cc @MrTimscampi as well, he might have some valuable input for this.

Reply:
> * In store action, do the api request and check with lodash's `isEqual` if the response is equal to what we have in the store. If it's equal, do nothing, if it isn't, commit a mutation, so the mappedState in point 1 gets an update and user views are changed to the user.

I made these changes to remove waiting for an initial network of userviews before displaying it in the home sections. By waiting, and comparing if the two are equal, removes this improvement, and won't improve the loading speeds)

Reply:
> * Fully reactive if we commit the refresh action on a websocket message somewhere else after initialization

Even if we listened to a websocket message and updated the state accordingly, because of how we currently generate home sections
(Get array of home sections, get userviews, iterate through userviews to get latest item sections, map sections in template) it still wouldn't be reactive